### PR TITLE
added a warning popup for login user who is inactive for 20 minutes

### PR DESCRIPTION
**Pull request by sufwankhalid1**
currently there is default 30 mintues of timeout if user is not active to handle this situation  a popup is added  which shows up after 20 mintues of inactivity and allow user to extend session 

**Changes**
- added endpoint SessionExtend to extend the current session
- added modal in header.jsp 
- added sessionWarningTime variable commonConfiguration

### CODE REVIEW

File: src/main/resources/bundles/es/header.properties
Comment (TanyaStere42) [inline src/main/resources/bundles/es/header.properties:1]: There's encoding issues here. Recommended to change to unicode escape sequence so display is correct

**Review by TanyaStere42 — CHANGES_REQUESTED** [review]

File: src/main/resources/bundles/es/header.properties
Comment (sufwankhalid1) [reply src/main/resources/bundles/es/header.properties:1]: thanks for pointing out i have updated it to unicode escape sequence

File: src/main/resources/bundles/es/header.properties [hunk 1]
```diff
@@ -92,7 +92,7 @@ dataIntegrity = Integridad de los datos
 manageProjects = Gestionar proyectos
 researchProjects = Proyectos de investigacion
 
-extendButton = Extender Sesi�n
+extendButton = Extender Sesi\\xf3n
```
Comment (TanyaStere42) [inline src/main/resources/bundles/es/header.properties:95]: I can't sort out what this character is. Is this a typo?

File: src/main/java/org/ecocean/CommonConfiguration.java [hunk 2]
```diff
@@ -1021,4 +1021,25 @@ public static String getWildbookCommunityURL(String context) {
...
+  /**
+   * Retrieves the sessionWarningTime.
+   *
+   * @param context Webapp context
+   * @return The sessionWarningTime
+   */
+
+    public static int getSessionWarningTime(String context) {
+
+    int def = 20;
```
Comment (naknomum) [inline src/main/java/org/ecocean/CommonConfiguration.java:1033]: can we leave a comment (for future reference) that this value is in minutes? maybe also in the properties file for good luck.

File: src/main/webapp/header.jsp [hunk 3]
```diff
@@ -190,6 +191,47 @@ finally{
...
+
+            // Session warning times in minutes
+            var warningTime = <%=sessionWarningTime %>;
+
+            function showWarning() {
+              $('#sessionModal').modal('show');
+            }
+
+            function extendSession() {
+                $.get("ExtendSession", function() {
```
Comment (naknomum) [inline src/main/webapp/header.jsp:208]: i am thinking this url should be _absolute_ (e.g. `/ExtendSession`), as there are pages (like `/encounters/encounter.jsp`) which will get a 404 if this is relative.

**Review by naknomum — CHANGES_REQUESTED** [review]
just a couple small notes. i think the `/ExtendSession` one is the main one.

File: src/main/resources/bundles/es/header.properties (hunk 1 above)
Comment (sufwankhalid1) [reply src/main/resources/bundles/es/header.properties]: yeah it was typo i have fixed it extendButton = Extender Sesi\u00F3n

File: src/main/java/org/ecocean/CommonConfiguration.java (hunk 2 above)
Comment (sufwankhalid1) [reply src/main/java/org/ecocean/CommonConfiguration.java:1033]: thanks for suggestion i have updated the docstring and added comment in properties

File: src/main/webapp/header.jsp (hunk 3 above)
Comment (sufwankhalid1) [reply src/main/webapp/header.jsp]: i have updated extendSession to use similar approach which is already being used in project kindly let me know if its good

Comment (naknomum) [reply src/main/webapp/header.jsp]: ah right, that global js variable seems like the better way to do it. thanks.

**Review by naknomum — APPROVED** [review]
looks good

**Review by TanyaStere42 — APPROVED** [review]
cool cool, lgtm

**Pull request merged**